code/game/machinery/tcomms/_base.dm

```diff
@@ -0,0 +1,442 @@
+		if(tcm.sender)
+			quotedmsg = "[tcm.sender.say_quote(multilingual_to_message(tcm.message_pieces))], \"[multilingual_to_message(tcm.message_pieces)]\""
+		else
+			quotedmsg = "says, \"[multilingual_to_message(tcm.message_pieces)]\""
+
+		// --- This following recording is intended for research and feedback in the use of department radio channels ---
+
+		var/part_blackbox_b = "</span><b> \[[freq_text]\]</b> <span class='message'>" // Tweaked for security headsets -- TLE
+		var/blackbox_msg = "[part_a][tcm.sender_name][part_blackbox_b][quotedmsg][part_c]"
+		//var/blackbox_admin_msg = "[part_a][M.name] (Real name: [M.real_name])[part_blackbox_b][quotedmsg][part_c]"
+
+		//BR.messages_admin += blackbox_admin_msg
+		if(istype(GLOB.blackbox))
+			switch(display_freq)
+				if(PUB_FREQ)
+					GLOB.blackbox.msg_common += blackbox_msg
+				if(SCI_FREQ)
+					GLOB.blackbox.msg_science += blackbox_msg
+				if(COMM_FREQ)
+					GLOB.blackbox.msg_command += blackbox_msg
+				if(MED_FREQ)
+					GLOB.blackbox.msg_medical += blackbox_msg
+				if(ENG_FREQ)
+					GLOB.blackbox.msg_engineering += blackbox_msg
+				if(SEC_FREQ)
+					GLOB.blackbox.msg_security += blackbox_msg
+				if(DTH_FREQ)
+					GLOB.blackbox.msg_deathsquad += blackbox_msg
+				if(SYND_FREQ)
+					GLOB.blackbox.msg_syndicate += blackbox_msg
+				if(SYNDTEAM_FREQ)
+					GLOB.blackbox.msg_syndteam += blackbox_msg
+				if(SUP_FREQ)
+					GLOB.blackbox.msg_cargo += blackbox_msg
+				if(SRV_FREQ)
+					GLOB.blackbox.msg_service += blackbox_msg
+				else
+					GLOB.blackbox.messages += blackbox_msg
+		//End of research and feedback code.
+
+	 /* ###### Send the message ###### */
+
+
+	  	/* --- Process all the mobs that heard a masked voice (understood) --- */
+
+		if(length(heard_masked))
+			for(var/mob/R in heard_masked)
+				R.hear_radio(tcm.message_pieces, tcm.verbage, part_a, part_b, tcm.sender, 0, tcm.sender_name, follow_target=tcm.follow_target)
+
+		/* --- Process all the mobs that heard the voice normally (understood) --- */
+
+		if(length(heard_normal))
+			for(var/mob/R in heard_normal)
+				R.hear_radio(tcm.message_pieces, tcm.verbage, part_a, part_b, tcm.sender, 0, tcm.sender_name, follow_target=tcm.follow_target)
+
+		/* --- Process all the mobs that heard the voice normally (did not understand) --- */
+
+		if(length(heard_voice))
+			for(var/mob/R in heard_voice)
+				R.hear_radio(tcm.message_pieces, tcm.verbage, part_a, part_b, tcm.sender,0, tcm.vname, follow_target=tcm.follow_target)
+
+		/* --- Process all the mobs that heard a garbled voice (did not understand) --- */
+			// Displays garbled message (ie "f*c* **u, **i*er!")
+
+		if(length(heard_garbled))
+			for(var/mob/R in heard_garbled)
+				R.hear_radio(tcm.message_pieces, tcm.verbage, part_a, part_b, tcm.sender, 1, tcm.vname, follow_target=tcm.follow_target)
+
+
+		/* --- Complete gibberish. Usually happens when there's a compressed message --- */
+
+		if(length(heard_gibberish))
+			for(var/mob/R in heard_gibberish)
```

Comment:
can there be anything other than mobs in this list? If not, make this istypeless.